### PR TITLE
Fixes to user invitation service

### PR DESCRIPTION
**Pull request by allenwq**
Due to a bug in rails (same with the one in duplication), the valid records get saved without rollback when there are invalid records.  Fixed by saving the records individually and rollback when there're errors.

### CODE REVIEW

File: config/locales/en/course/user_invitations.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ en:
     user_invitations:
       errors:
         duplicate_user: '%{user} appears more than once in the submission.'
-        invalid_email: '%{email} is an invalid email address.'
+        invalid_email: "%{email} is invalid or already taken."
```
Comment (fonglh) [inline config/locales/en/course/user_invitations.yml]: Can we tell the difference between invalid and already taken easily?

Comment (allenwq) [inline config/locales/en/course/user_invitations.yml]: Need to implement a separate validation for it.

Comment (allenwq) [inline config/locales/en/course/user_invitations.yml]: @fonglh Have made it to show the specific message in the second commit. now the second part of the message is generated by rails, in the format of `xxx@emai.com is invliad: Email has already been take`. Any suggestions?

Comment (fonglh) [inline config/locales/en/course/user_invitations.yml]: That's fine, are there other ways it could be invalid?

Comment (allenwq) [inline config/locales/en/course/user_invitations.yml]: other cases are: email format invalid, email address too long, etc.

**Pull request merged**